Fabric.Authorization.API/scripts/AzureADUsers-Migration.ps1

```diff
@@ -0,0 +1,390 @@
+$identityInstallUtilities = ".\Install-Identity-Utilities.psm1"
+if (!(Test-Path $identityInstallUtilities -PathType Leaf)) {
+    Write-DosMessage -Level "Warning" -Message "Could not find identity install utilities. Manually downloading and installing"
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/HealthCatalyst/Fabric.Identity/master/Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1 -Headers @{"Cache-Control" = "no-cache"} -OutFile $identityInstallUtilities
+}
+Import-Module -Name $identityInstallUtilities -Force
+
+# Import Fabric Install Utilities
+$fabricInstallUtilities = ".\Fabric-Install-Utilities.psm1"
+if (!(Test-Path $fabricInstallUtilities -PathType Leaf)) {
+    Write-DosMessage -Level "Warning" -Message "Could not find fabric install utilities. Manually downloading and installing"
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/HealthCatalyst/InstallScripts/master/common/Fabric-Install-Utilities.psm1 -Headers @{"Cache-Control" = "no-cache"} -OutFile $fabricInstallUtilities
+}
+Import-Module -Name $fabricInstallUtilities -Force
+$credentials = @{}
+
+function Get-AuthUsers
+{	
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $connectionString
+    )
+    $connection = New-Object System.Data.SqlClient.SqlConnection($connectionString)	
+    $sql = "SELECT u.Id, u.SubjectId, u.IdentityProvider, u.IdentityProviderUserPrincipalName, g.GroupId, r.RoleId, p.PermissionId, p.PermissionAction, CAST('' AS NVARCHAR(100)) AS 'sourceAnchor', CAST('' AS NVARCHAR(100)) AS 'objectId'
+            FROM [dbo].[Users] u	
+            LEFT OUTER JOIN [dbo].[GroupUsers] g	
+                ON u.SubjectId = g.SubjectId AND u.IdentityProvider = g.IdentityProvider
+		    LEFT OUTER JOIN [dbo].[RoleUsers] r
+                ON u.SubjectId = r.SubjectId AND u.IdentityProvider = r.IdentityProvider
+            LEFT OUTER JOIN [dbo].[UserPermissions] p	
+                ON u.SubjectId = p.SubjectId AND u.IdentityProvider = p.IdentityProvider
+            WHERE u.IsDeleted = 0
+				AND u.IdentityProvider = 'Windows'
+				AND u.IdentityProviderUserPrincipalName IS NOT NULL";	
+    $command = New-Object System.Data.SqlClient.SqlCommand($sql, $connection)	
+
+    try {	
+        $connection.Open()    	
+        $result = $command.ExecuteReader()	
+        $userTable = New-Object System.Data.DataTable # Create a DataTable
+        $userTable.Load($result) # Load the results from the DataReader into the DataTable
+        $userTable.Columns['sourceAnchor'].ReadOnly = $false
+        $userTable.Columns['objectId'].ReadOnly = $false
+        $connection.Close()        	
+    }	
+    catch [System.Data.SqlClient.SqlException] {
+        Write-DosMessage -Level "Fatal" -Message "An error ocurred while executing the command. Please ensure the connection string is correct and the authorization database has been setup. Connection String: $($connectionString). Error $($_.Exception.Message)"
```

Comment:
Do we not want to print out `$_.Exception` vs. `$_.Exception.Message`?

Reply:
Exception.ToString() would give us more verbose output, including a stack trace, which would be more valuable for debugging. I was just following what we are doing currently.

Reply:
I think we're doing `$_.Exception` in some places but using just `.Message` in other places.

Reply:
I don't think you should need the `ToString()`.

Reply:
Maybe not in powershell, that is just the documentation I have come across in .Net framework.